server/src/routes/series.js

```diff
@@ -0,0 +1,126 @@
+    } catch (e) {
+        statusCode = HttpStatus.INTERNAL_SERVER_ERROR;
+        response = `Failed while fetching genres: ${e}`;
+    }
+
+    res.status(statusCode).send(response);
+});
+
+router.get('/filters', async (req, res) => {
+    let genres;
+    let statusCode = HttpStatus.OK;
+    const statuses = Series.schema.path('status').enumValues;
+    try {
+        genres = await Genres.find();
+    } catch (e) {
+        statusCode = HttpStatus.INTERNAL_SERVER_ERROR;
+        genres = `Failed while fetching genres: ${e}`;
+    }
+
+    res.status(statusCode).send({ genres, statuses });
+});
+
+router.get('/commonAmongFollowing/:email', async (req, res) => {
+    const { pageNumber } = req.query;
+    const { email } = req.params;
+
+    let response;
+    let statusCode = HttpStatus.OK;
+    try {
+        const userWatchList = await getUserWatchlist(email);
+        response = await getCommonSeriesAmongFollowing(email, userWatchList.map(series => series.name), pageNumber, pageLimit);
+    } catch (e) {
+        statusCode = HttpStatus.INTERNAL_SERVER_ERROR;
+        response = `Failed while fetching common series amoung following: ${e}`;
+    }
+
+    res.status(statusCode).send(response);
+});
+
+router.get('/watched', async (req, res) => {
+    const { pageNumber } = req.query;
+
+    let response;
+    let statusCode = HttpStatus.OK;
+    try {
+        response = await getMostWatchedSeries(pageNumber, pageLimit);
+    } catch (e) {
+        statusCode = HttpStatus.INTERNAL_SERVER_ERROR;
+        response = `Failed while fetching most watched series: ${e}`
+        console.log(response);
+    }
+
+    res.status(statusCode).send(response);
+
+});
+
+router.get('/topRated', async (req, res) => {
+    const { pageNumber } = req.query;
+
+    let response;
+    let statusCode = HttpStatus.OK;
+    try {
+        response = await Series.find().sort({ vote_average: -1 }).skip(pageLimit * (pageNumber - 1)).limit(pageLimit).exec();
+    } catch (e) {
+        statusCode = HttpStatus.INTERNAL_SERVER_ERROR;
+        response = `Failed while fetching series by rating: ${e}`
+        console.log(response);
+    }
+
+    res.status(statusCode).send(response);
+});
+
+router.get('/popular', async (req, res) => {
+    const { pageNumber } = req.query;
+
+    let response;
+    let statusCode = HttpStatus.OK;
+    try {
+        response = await Series.find().sort({ popularity: -1 }).skip(pageLimit * (pageNumber - 1)).limit(pageLimit).exec();
+    } catch (e) {
+        statusCode = HttpStatus.INTERNAL_SERVER_ERROR;
+        response = `Failed while fetching series by popularity: ${e}`;
+        console.log(response);
+    }
+
+    res.status(statusCode).send(response);
+});
+
+router.get('/:id', async (req, res) => {
+    const { id } = req.params;
+
+    let response;
+    let statusCode = HttpStatus.OK;
+    try {
+        response = await aggregateSeries([new mongoose.mongo.ObjectId(id)]);
```

Comment:
Please review what I did in this same route in my branch (series details).
This response returns an array whereas my response returns an object. If the id doesn't exist, it returns [] which is truthy => status code OK.
Also, it's implemented in `series.js` service in my branch.